pylot/perception/detection/detection_operator.py

```diff
@@ -38,42 +39,14 @@ def __init__(self, camera_stream, time_to_decision_stream,
         self._flags = flags
         self._logger = erdos.utils.setup_logging(self.config.name,
                                                  self.config.log_file_name)
-        self._detection_graph = tf.Graph()
-        # Load the model from the model file.
-        pylot.utils.set_tf_loglevel(logging.ERROR)
-        with self._detection_graph.as_default():
-            od_graph_def = tf.GraphDef()
-            with tf.gfile.GFile(model_path, 'rb') as fid:
-                serialized_graph = fid.read()
-                od_graph_def.ParseFromString(serialized_graph)
-                tf.import_graph_def(od_graph_def, name='')
-
-        self._gpu_options = tf.GPUOptions(
-            allow_growth=True,
-            visible_device_list=str(self._flags.obstacle_detection_gpu_index),
-            per_process_gpu_memory_fraction=flags.
-            obstacle_detection_gpu_memory_fraction)
-        # Create a TensorFlow session.
-        self._tf_session = tf.Session(
-            graph=self._detection_graph,
-            config=tf.ConfigProto(gpu_options=self._gpu_options))
-        # Get the tensors we're interested in.
-        self._image_tensor = self._detection_graph.get_tensor_by_name(
-            'image_tensor:0')
-        self._detection_boxes = self._detection_graph.get_tensor_by_name(
-            'detection_boxes:0')
-        self._detection_scores = self._detection_graph.get_tensor_by_name(
-            'detection_scores:0')
-        self._detection_classes = self._detection_graph.get_tensor_by_name(
-            'detection_classes:0')
-        self._num_detections = self._detection_graph.get_tensor_by_name(
-            'num_detections:0')
+
+        _model_dir = (os.sep).join(model_path.split(os.sep)[:-1])
```

Comment:
The more canonical way here is to do `os.path.join` and `os.path.basename`

Reply:
Can you elaborate why you needed to do the path manipulation because the earlier iteration of the code took as input the path to the frozen protobuff of the model?